design/src/rfcs/rfc0012_crates_io_publishing.md

```diff
@@ -0,0 +1,214 @@
+
+Cargo uses [semver] for versioning, with a `major.minor.patch-pre` format:
+- `major`: Incompatible API changes
+- `minor`: Added functionality in backwards compatible manner
+- `patch`: Backwards compatible bug fixes
+- `pre`: Pre-release version tag (omitted for normal releases)
+
+In the new versioning strategy, the `minor` version number will no longer be coordinated across
+all SDK and Smithy runtime crates.
+
+During phases 1 and 2, the `major` version will always be 0, and the following scheme will be used:
+- `minor`:
+  - New features
+  - Breaking changes
+  - Dependency updates for dependencies that are part of the public API
+  - Model updates with API changes
+  - For code-generated crates: when a newer version of [smithy-rs] is used to generate the crate
+- `patch`:
+  - Bug fixes that do not break backwards compatibility
+  - Model updates that _only_ have documentation changes
+
+During phase 3:
+- `major`: Breaking changes
+- `minor`:
+    - Changes that aren't breaking
+    - Dependency updates for dependencies that are part of the public API
+    - Model updates with API changes
+    - For code-generated crates: when a newer version of [smithy-rs] is used to generate the crate
+- `patch`:
+    - Bug fixes that do not break backwards compatibility
+    - Model updates that _only_ have documentation changes
+
+During phase 3, bumps to the `major` version must be coordinated across all SDK and runtime crates.
+
+### Release Identification
+
+Since there will no longer be one SDK "version", release tags will be dates in `YYYY-MM-DD` format
+rather than version numbers. Additionally, the SDK's user agent string will need to include a separate
+service version number (this requirement has already been implemented).
+
+### Yanking
+
+It must be possible to yank an entire release with a single command-line. The publisher tool must
+be updated to understand which crate versions were released with a given release tag, and be able to
+yank that entire tag worth of crates.
+
+Phase 1: New Versioning Strategy
+--------------------------------
+
+Phase 1 will address the following challenges introduced by uncoordinating the `major.minor` versions:
+- Tracking of versions associated with a release tag
+- Creation of version bump process for code generated crates
+- Enforcement of version bump process in runtime crates
+- Yanking of versions associated with a release tag
+
+### Version Tracking
+
+A new manifest file will be introduced in the root of [aws-sdk-rust] named `versions.toml` that describes
+all versioning information for any given commit in the repository. In the main branch, the `versions.toml`
+in tagged commits will become the source of truth for which crate versions belong to that release, as well
+as additional metadata that's required for maintaining version process in the future.
+
+This format will look as follows:
+```toml
+smithy_rs_version = "<release-tag|commit-hash>"
+
+[aws-smithy-types]
+version = "0.50.1"
```

Comment:
the runtime crate versions here are just for consistency right? they aren't actually required for the system to function?

We _could_ consider tracking `published = ...` and `next = ...` in this file which removes the crates.io API dependency (or would just provide a spot to verify)

Reply:
The idea was to have all the version numbers for a given release tag centralized in one place, so that's why the runtime crate versions are in here. Nothing else is requiring them to be in this file.